full-service/src/service/sync.rs

```diff
@@ -477,4 +598,66 @@ mod tests {
             .expect("Could not get balance");
         assert_eq!(balance.unspent, 250_000_000 * MOB as u128);
     }
+
+    #[test_with_logger]
+    fn test_sync_view_only_account(logger: Logger) {
+        let mut rng: StdRng = SeedableRng::from_seed([20u8; 32]);
+
+        let view_private_key = RistrettoPrivate::from_random(&mut rng);
+        // let key =
+        //     hex::decode("
+        // 0a20928c29f916586c0fae22de17784b2b9ac573a1b1d75c2ba531838650ca0a5302")
+        //         .unwrap();
+        // let view_private_key = mc_util_serial::decode(&key).unwrap();
+
+        let spend_private_key = RistrettoPrivate::from_random(&mut rng);
+
+        let account_key = AccountKey::new(&spend_private_key, &view_private_key);
+
+        let mut ledger_db = get_test_ledger(0, &vec![], 0, &mut rng);
+
+        let origin_block_amount: u128 = 250_000_000 * MOB as u128;
+        let origin_block_txo_amount = origin_block_amount / 16;
+        let o = account_key.subaddress(0);
+        let _new_block_index = add_block_to_ledger_db(
+            &mut ledger_db,
+            &[
+                o.clone(),
+                o.clone(),
+                o.clone(),
+                o.clone(),
+                o.clone(),
+                o.clone(),
+                o.clone(),
+                o.clone(),
+                o.clone(),
+                o.clone(),
+                o.clone(),
+                o.clone(),
+                o.clone(),
+                o.clone(),
+                o.clone(),
+                o.clone(),
+            ],
```

Comment:
Optional/Equivalent: Use [`Range.map`](https://doc.rust-lang.org/std/ops/struct.Range.html#method.map) to DRY these calls.
```suggestion
            &(0..15).map(|_| o.clone()).collect::<Vec<_>>(),
```